lib/obsolete.gd

```diff
@@ -76,19 +76,29 @@
 ##  <#/GAPDoc>
 ##
 
-BIND_GLOBAL( "DeclareObsoleteSynonym", function( name_obsolete, name_current )
-    local value, orig_value;
+BIND_GLOBAL( "DeclareObsoleteSynonym", function( name_obsolete, name_current, level_arg... )
+    local value, orig_value, printed_warning, level;
     if not ForAll( [ name_obsolete, name_current ], IsString ) then
         Error("Each argument of DeclareObsoleteSynonym must be a string\n");
     fi;
+    if Length(level_arg) = 0 then
+        level := 2;
+    else
+        level := level_arg[1];
+    fi;
+
     value := EvalString( name_current );
     if IsFunction( value ) then
         orig_value := value;
+        printed_warning := false;
         value := function (arg)
             local res;
-            Info( InfoObsolete, 1, "'", name_obsolete, "' is obsolete.",
-                "\n#I  It may be removed in a future release of GAP.",
-                "\n#I  Use ", name_current, " instead.");
+            if not printed_warning then
```

Comment:
It seems this change also causes each Info message to be shown only once. That seems reasonable to me, but it should be mentioned in the commit message and the PR description.